### ISSUE TITLE
`env` attribute not compatible with `lazy`

### ISSUE THREAD

**Issue by mfischer-zd** [chef-cookbooks/runit#202]
You can't currently use a `lazy` block with the `env` attribute.

Example declaration:

```
runit_service 'foo' do
  env lazy { some_hash }
end
```

```
================================================================================
Error executing action `enable` on resource 'runit_service[xxxx]'
================================================================================

NoMethodError
-------------
undefined method `empty?' for #<Chef::DelayedEvaluator:0x00000002a706c8>

Cookbook Trace:
---------------
/tmp/kitchen/cache/cookbooks/runit/libraries/resource_runit_service.rb:130:in `options'
/tmp/kitchen/cache/cookbooks/runit/libraries/provider_runit_service.rb:88:in `block (2 levels) in <class:RunitService>'
/tmp/kitchen/cache/cookbooks/runit/libraries/provider_runit_service.rb:82:in `block in <class:RunitService>'
/tmp/kitchen/cache/cookbooks/runit/libraries/provider_runit_service.rb:280:in `block in <class:RunitService>'
```

**Comment by wk8** [comment]
Just hit that too, +1

**Comment by tas50** [comment]
This is a known issue we have with several of the options since we merge in some default values with the values that are passed. I'll see if there's a way we can do this that would work better with lazy.

**Comment by tas50** [comment]
There's a distinct possibility that this if fixed by #241

**Comment by tas50** [comment]
I'm closing this out since I believe the refactor solves it
